plugin-test/src/test/groovy/org/gradle/github/dependency/extractor/MultiProjectDependencyExtractorTest.groovy

```diff
@@ -1,370 +1,242 @@
-            dependencies == [this.fooGav]
-        }
-
-        def cCompileClasspath = jsonManifest("project :c")
-        def cCompileClasspathFile = cCompileClasspath.file as Map
-        cCompileClasspathFile.source_location == "c/build.gradle"
-        def cClasspathResolved = cCompileClasspath.resolved as Map
-        verifyAll(cClasspathResolved[fooGav] as Map) {
-            package_url == this.fooPurl
-            relationship == "indirect"
-            dependencies == []
-        }
-        verifyAll(cClasspathResolved["project :a"] as Map) {
-            package_url == aTestProjectPurl
-            relationship == "indirect"
-            dependencies == [this.fooGav]
-        }
-        verifyAll(cClasspathResolved["project :b"] as Map) {
-            package_url == bTestProjectPurl
-            relationship == "direct"
-            dependencies == ["project :a"]
-        }
-        where:
-        taskInvocation | _
-        "classes"      | _
-        ":c:classes"   | _
+        manifestNames == ["project :a", "project :b", "project :c"]
+
+        def manifestA = gitHubManifest("project :a")
+        manifestA.sourceFile == "a/build.gradle"
+        manifestA.assertResolved([
+            "org.test:foo:1.0": [package_url: fooPurl]
+        ])
+
+        def manifestB = gitHubManifest("project :b")
+        manifestB.sourceFile == "b/build.gradle"
+        manifestB.assertResolved([
+            "project :a"      : [
+                package_url : "pkg:maven/org.test/a@1.0",
+                dependencies: ["org.test:foo:1.0"]
+            ],
+            "org.test:foo:1.0": [
+                package_url : fooPurl,
+                relationship: "indirect"
+            ]
+        ])
+
+        def manifestC = gitHubManifest("project :c")
+        manifestC.sourceFile == "c/build.gradle"
+        manifestC.assertResolved([
+            "project :b"      : [
+                package_url : "pkg:maven/org.test/b@1.0",
+                dependencies: ["project :a"]
+            ],
+            "project :a"      : [
+                package_url : "pkg:maven/org.test/a@1.0",
+                relationship: "indirect",
+                dependencies: ["org.test:foo:1.0"]
+            ],
+            "org.test:foo:1.0": [
+                package_url : fooPurl,
+                relationship: "indirect"
+            ]
+
+        ])
     }
 
-    def "project with buildSrc"() {
+    def "extracts dependencies from buildSrc project"() {
         given:
-        multiProjectBuild("parent", []) {
-            project("buildSrc").tap {
-                buildFile """
-                apply plugin: 'java'
-                group = 'org.test.buildSrc'
-                version = '1.0'
-                dependencies {
-                    implementation 'org.test-published:foo:1.0'
-                }
-                """
-                setupBuildFile(it)
+        file("buildSrc/settings.gradle") << "rootProject.name = 'buildSrc'"
```

Comment:
Is a `rootProject.name` required here? I believe we encourage it as best practice, but I'm wondering if it has any actual relevance to the test.

Reply:
No, `rootProject.name` is not required (in either top-level `settings.gradle` or `buildSrc/settings.gradle`. 
But this provides some basic content to use when creating the settings file (which is mandatory for top-level projects and included builds, but optional for `buildSrc`).